### PR TITLE
Added heading to Motioncommand2d

### PR DESCRIPTION
**Pull request by Cirromulus**
To support omnidirectional drives I suggest adding a heading value to Motion2D-command.
(As suggested in comment of previous committer)

### CODE REVIEW

**Comment by doudou** [comment]
MotionCommand2D represent velocities, not positions. I don't see how the heading is relevant.

**Comment by Cirromulus** [comment]
Heading is relevant for omnidirectional drives, e.g for driving sideways. Furthermore it enables the device doing circles facing in any other direction than the driving-vector as well.
Heading is an angle that rotates the driving vector.

Eg.: 
Sideways would be [translation = 1, rotation = 0, heading = 90°]
Circling around a point in front of Device facing towards point would be [translation = 1, rotation = 1, heading -90°]

**Comment by doudou** [comment]
OK ... got it. It's basically the direction of the translation vector w.r.t. the X axis of the vehicle.

My only comment then would be to use a base::Angle.

**Comment by Cirromulus** [comment]
Ok, i'll change it

**Comment by Cirromulus** [comment]
Better?

File: base/commands/Motion2D.hpp [hunk 1]
```diff
@@ -13,9 +15,11 @@ namespace base
     {
         double translation; //! translation value in m/s
         double rotation;    //! rotation value in rad/s. Positive is counter-clockwise
-        double heading;     //! heading in rad. Positive is counter-clockwise
-        Motion2D():translation(0), rotation(0), heading(0){};
-        Motion2D(double translation, double rotation, double heading):translation(translation), rotation(rotation), heading(heading){};
+        base::Angle heading;     //! heading in rad. Positive is counter-clockwise
+        
+        Motion2D():translation(0), rotation(0), heading(base::Angle::fromRad(0)){};
+        Motion2D(double translation, double rotation):translation(translation), rotation(rotation), heading(base::Angle::fromRad(0)){};
```
Comment (jmachowinski) [inline base/commands/Motion2D.hpp]: Better use the default constructor. 
I initializes with base::unset<double>. In this case 
one can distinguish between not used and Zero..

**Comment by Cirromulus** [comment]
In this Construct Zero is equivalent to unset, because an Angle of 0 would not shift the Vector, as this would be the case before it was introduced.

**Comment by jmachowinski** [comment]
Ok

**Pull request merged**